### PR TITLE
Create Kid Interface

### PR DESCRIPTION
**Pull request by GovindMehra**

### CODE REVIEW

**Comment by GovindMehra** [comment]
<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>My Chores</title>
    <style>
        body {
            font-family: Arial, sans-serif;
            margin: 0;
            padding: 0;
            box-sizing: border-box;
        }
        header {
            background-color: #f5d93a;
            color: red;
            padding: 10px 0;
            display: flex;
            align-items: center;
        }
        header img {
            margin-left: 20px;
        }
        header h1 {
            flex-grow: 1;
            text-align: center;
            margin: 0;
        }
        nav {
            display: flex;
            justify-content: center;
            background-color: #bca62c;
        }
        nav a {
            color: white;
            padding: 14px 20px;
            text-decoration: none;
            text-align: center;
        }
        nav a:hover {
            background-color: red;
            color: black;
        }
        .container {
            padding: 10px;
            background-color: #FF3131;
        }
        table {
            width: 100%;
            border-collapse: collapse;
            margin-top: 20px;
        }
        th, td {
            border: 2px solid black;
            padding: 10px;
            text-align: center;
        }
        th {
            background-color: #50357d;
            color: white;
        }
        .complete-button {
            background-color: #50357d;
            color: white;
            border: none;
            cursor: pointer;
            padding: 8px 16px;
            border-radius: 4px;
        }
        .complete-button:hover {
            background-color: #311f4d;
        }
    </style>
</head>
<body>
    <header>
        SoulHeart Logo
        <h1>Welcome to SoulHeart</h1>
    </header>
    <nav>
        <a href="./homeK.html">Home</a>
        <a href="./choresK.html">My Chores</a>
        <a href="./rewardsK.html">My Rewards</a>
    </nav>
    <div class="container">
        <h2>My Chores</h2>
        <table id="choreTable">
            <thead>
                <tr>
                    <th>Chore Name</th>
                    <th>Status</th>
                    <th>No. of Stars</th>
                    <th>Deadline</th>
                    <th>Action</th>
                </tr>
            </thead>
            <tbody>
                
            </tbody>
        </table>
    </div>

    <script>
        // Simulated initial data - replace with actual data fetch logic
        let chores = [
            { choreName: "Wash dishes", status: "Incomplete", stars: 3,
deadline: "2024-07-20" },
            { choreName: "Clean room", status: "Incomplete", stars: 5,
deadline: "2024-07-15" },
            { choreName: "Take out trash", status: "Incomplete", stars: 2,
deadline: "2024-07-18" }
        ];

        // Function to fetch chore data and populate table
        function fetchChores() {
            const tbody = document.querySelector('#choreTable tbody');
            tbody.innerHTML = ''; // Clear existing rows

            // Append all chores to the table
            chores.forEach(chore => {
                appendChoreRow(chore);
            });
        }

        // Function to add a chore row to the table
        function appendChoreRow(chore) {
            const tbody = document.querySelector('#choreTable tbody');
            const row = document.createElement('tr');
            row.innerHTML = `
                <td>${chore.choreName}</td>
                <td>${chore.status}</td>
                <td>${chore.stars}</td>
                <td>${chore.deadline}</td>
                <td>${getActionButton(chore)}</td>
            `;
            tbody.appendChild(row);
        }

        // Function to determine the action button based on chore status
and deadline
        function getActionButton(chore) {
            if (chore.status === 'Complete' || isDeadlinePassed(chore.
deadline)) {
                return '';
            } else {
                return `<button class="complete-button"
onclick="markAsComplete(this)">Mark as Complete</button>`;
            }
        }

        // Function to check if deadline is passed
        function isDeadlinePassed(deadline) {
            const today = new Date();
            const choreDeadline = new Date(deadline);
            return today > choreDeadline;
        }

        // Function to mark chore as complete
        function markAsComplete(button) {
            const row = button.parentNode.parentNode;
            const choreIndex = Array.from(row.parentNode.children).indexOf(
row);

            // Update chore status to Complete or Failed to Complete
            if (isDeadlinePassed(chores[choreIndex].deadline)) {
                chores[choreIndex].status = 'Failed to Complete';
            } else {
                chores[choreIndex].status = 'Complete';
            }

            // Update status cell in the row
            row.cells[1].textContent = chores[choreIndex].status;

            // Remove action button from completed or failed chore
            row.cells[4].innerHTML = '';

            // Move completed or failed chore to the bottom of the list
            const tbody = document.querySelector('#choreTable tbody');
            tbody.appendChild(row);
        }

        // Fetch chores when page loads
        fetchChores();
    </script>
</body>
</html>

**Comment by ShrotGarg** [comment]
<!DOCTYPE html>
<html lang="en">
<head>
<meta charset="UTF-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Chore List</title>
<style>
body {
font-family: Arial, sans-serif;
margin: 0;
padding: 0;
box-sizing: border-box;
}
header {
background-color: #31234A;
color: red;
padding: 10px 0;
display: flex;
align-items: center;
}
header img {
margin-left: 20px;
}
header h1 {
flex-grow: 1;
text-align: center;
margin: 0;
}
nav {
display: flex;
justify-content: center;
background-color: #50357d;
}
nav a {
color: white;
padding: 14px 20px;
text-decoration: none;
text-align: center;
}
nav a:hover {
background-color: red;
color: black;
}
.container {
position: relative;
padding: 10px;
background-color: #FF3131;
}
table {
width: 100%;
border-collapse: collapse;
margin-top: 20px;
}
th, td {
border: 2px solid black;
padding: 10px;
text-align: center;
}
th {
background-color: #50357d;
color: white;
}
.add-button {
position: absolute;
top: 10px;
right: 10px;
background-color: #50357d;
color: white;
border: none;
padding: 10px 20px;
cursor: pointer;
font-size: 20px;
border-radius: 50%;
}
.delete-button {
background-color: transparent;
border: none;
cursor: pointer;
font-size: 20px;
color: black;
}
.form-popup {
display: none;
position: fixed;
bottom: 20px;
right: 20px;
border: 3px solid #f1f1f1;
z-index: 9;
background-color: white;
padding: 20px;
box-shadow: 0px 0px 10px 0px rgba(0, 0, 0, 0.1);
}
.form-container {
max-width: 300px;
padding: 10px;
background-color: white;
}
.form-container input[type=text], .form-container input[type=date], .
form-container input[type=number] {
width: 100%;
padding: 15px;
margin: 5px 0 10px 0;
border: none;
background: #f1f1f1;
}
.form-container input[type=text]:focus, .form-container input[type=date]:
focus, .form-container input[type=number]:focus {
background-color: #ddd;
outline: none;
}
.form-container .btn {
background-color: #50357d;
color: white;
border: none;
cursor: pointer;
width: 100%;
margin-bottom:10px;
opacity: 0.8;
padding: 15px;
}
.form-container .btn:hover {
opacity: 1;
}
.form-container .cancel {
background-color: red;
}
</style>
</head>
<body>
<header>
SoulHeart Logo
<h1>Welcome to SoulHeart</h1>
</header>
<nav>
<a href="{% url 'parents_dashboard' %}">Home</a>
<a href="{% url 'add_chore' %}" onclick="openForm(); return false;">Add
Chores</a>
</nav>
<div class="container">
<h2>Chore List</h2>
<button class="add-button" onclick="openForm()">+</button>
<table>
<thead>
<tr>
<th>Kid name</th>
<th>Chore name</th>
<th>Status</th>
<th>No. of stars</th>
<th>Deadline</th>
<th>Action</th>
</tr>
</thead>
<tbody>

{% for chore in chores %}
<tr>
<td>{{ chore.kid_name }}</td>
<td>{{ chore.chore_name }}</td>
<td>{{ chore.status }}</td>
<td>{{ chore.stars }}</td>
<td>{{ chore.deadline }}</td>
<td><button class="delete-button" onclick="deleteChore(this, '{{ chore.id
}}')">🗑️</button></td>
</tr>
{% endfor %}
</tbody>
</table>
</div>


<div class="form-popup" id="choreForm">
<form method="post" class="form-container" id="addChoreForm" action="{% url
'add_chore' %}">
{% csrf_token %}
<h2>Add Chore</h2>

<label for="kidName"><b>Kid Name</b></label>
<input type="text" id="kidName" name="kid_name" required>

<label for="choreName"><b>Chore Name</b></label>
<input type="text" id="choreName" name="chore_name" required>

<label for="stars"><b>No. of Stars</b></label>
<input type="number" id="stars" name="stars" required>

<label for="deadline"><b>Deadline</b></label>
<input type="date" id="deadline" name="deadline" required>

<button type="submit" class="btn">Add Chore</button>
<button type="button" class="btn cancel" onclick="closeForm()">Close</button
>
</form>
</div>

<script src="
https://ajax.googleapis.com/ajax/libs/jquery/3.5.1/jquery.min.js"></script>
<script>
// Function to open the form
function openForm() {
document.getElementById("choreForm").style.display = "block";
}

// Function to close the form
function closeForm() {
document.getElementById("choreForm").style.display = "none";
}

// Function to delete a chore via AJAX
function deleteChore(button, choreId) {
const row = button.parentNode.parentNode;

$.ajax({
url: `/delete_chore/${choreId}/`,
type: 'POST',
data: {
csrfmiddlewaretoken: '{{ csrf_token }}'
},
success: function(response) {
console.log('Response:', response);
// Assuming the chore is successfully deleted, remove the row from the table
row.remove();
alert('Chore deleted successfully.');
// Redirect to chore_list page after successful deletion
window.location.href = '/chore_list';
},
error: function(xhr, status, error) {
console.error('Failed to delete chore:', error);
alert('Failed to delete chore.');
}
});
}
</script>
</body>
</html>
